config/routes.rb

```diff
@@ -1,7 +1,16 @@
 Rails.application.routes.draw do
+  devise_for :users
+  get 'likes/new'
+  get 'likes/create'
+  get 'new/create'
+  get 'comments/new'
+  get 'comments/create'
   root "users#index"
   resources :users, only: [:index, :show] do
-    resources :posts, only: [:index, :show]
+    resources :posts, only: [:index, :show, :new, :create] do
+      resources :comments, only: [:new, :create]
+      resources :likes, only: [:new, :create]
+    end
```

Comment:
[OPTIONAL] For a better experience, kindly ensure a user is welcomed with the sign_in page when the app is open as this can lead to a better user experience👍🏼.